### PR TITLE
Broker conf file suffix support

### PR DESCRIPTION
**Pull request by didrocks**
We want all broker configuration files to ends up with .conf. Adapt and update the API for it.
Ensure that local broker ID is Local, and that broker id is based on broker name and not its config file.

### CODE REVIEW

**Review by denisonbarbosa — APPROVED** [review]
LGTM! It's way better to have a known extension for the configuration file!

File: internal/brokers/broker.go [hunk 1]
```diff
@@ -67,32 +67,30 @@ type fieldValidator struct {
...
+		name = LocalBrokerName
+		id = LocalBrokerName
+	} else {
+		log.Debugf(ctx, "Loading broker from %q", configFile)
+		broker, name, brandIcon, err = newDbusBroker(ctx, bus, configFile)
 		if err != nil {
 			return Broker{}, err
 		}
+		h := fnv.New32a()
+		h.Write([]byte(name))
```
Comment (3v1n0) [inline internal/brokers/broker.go:87]: Worth mentioning that by design it never reruns an error with Hash32?

File: internal/brokers/broker_test.go [hunk 2]
```diff
@@ -68,11 +68,13 @@ func TestNewBroker(t *testing.T) {
 			if tc.wantErr {
 				configDir = filepath.Join(brokerConfFixtures, "invalid_brokers")
 			}
-			if tc.configFile != "" {
+			if tc.configFile == "-" {
+				tc.configFile = ""
+			} else if tc.configFile != "" {
```
Comment (3v1n0) [inline internal/brokers/broker_test.go:73]: What about using a `ptrValue` here so that we don't have to use the special `-` and this logic?

**Review by 3v1n0 — COMMENTED** [review]
LGTM, some suggested cleanups

**Comment by 3v1n0** [comment]
Ouch I had the diff opened and didn't notice it hot merged already... Well, not a big deal I suppose

File: internal/brokers/broker.go [hunk 3]
```diff
@@ -67,32 +67,30 @@ type fieldValidator struct {
...
 	var broker brokerer
-	var brandIcon string
-	log.Debugf(ctx, "Loading broker %q", name)
-	if configFile != "" {
-		broker, fullName, brandIcon, err = newDbusBroker(ctx, bus, configFile)
+	var id string
+	if configFile == "" {
+		name = LocalBrokerName
+		id = LocalBrokerName
+	} else {
```
[the remark this answers is not in the document]
Comment (didrocks) [reply internal/brokers/broker.go:80]: Good suggestion, I will do that one.

File: internal/brokers/broker.go (hunk 1 above)
Comment (didrocks) [reply internal/brokers/broker.go:87]: yeah, let’s make that comment explicit.

File: internal/brokers/broker_test.go (hunk 2 above)
Comment (didrocks) [reply internal/brokers/broker_test.go:73]: I don’t like the `ptrValue` for this. I prefer our existing pattern and this is what we are using in most of project. So let’s use that one :)

**Pull request merged**